crates/proto/src/rr/domain/label.rs

```diff
@@ -224,14 +224,28 @@ impl Borrow<[u8]> for Label {
     }
 }
 
-fn is_safe_ascii(c: char, is_first: bool, for_encoding: bool) -> bool {
+fn is_valid_ascii(c: char, is_first: bool, for_encoding: bool) -> bool {
+    if !is_safe_ascii(c) {
+        return false;
+    }
+
+    if is_first && c == '-' {
+        false
+    } else if for_encoding && c == '.' {
+        false
+    } else {
+        true
+    }
```

Comment:
just noting `clippy` doesn't like the `false` duplicate conditional bodies and this will need a tweak for CI to pass.